app/templates/settings/general.hbs

```diff
@@ -71,14 +71,18 @@
             </div>
             <div class="gh-setting-action gh-setting-action-smallimg">
                 {{#if model.icon}}
-                    <img class="blog-icon" src="{{model.icon}}" alt="icon" data-test-icon-img>
-                    <button type="button" class="gh-setting-action-smallimg-delete" {{action "removeImage" "icon"}} data-test-delete-image="icon">
-                        <span>delete</span>
-                    </button>
+                    {{#if uploader.isUploading}}
```

Comment:
Rather than duplicating the `if` block and progress bar it may be simpler to re-organise:

```
if uploader.isUploading
    progress
else if model.icon
    icon
else
     upload button
```